### PR TITLE
Render completed surveys and non - repetitive survey numbers and ques…

### PR DESCRIPTION
**Pull request by shahsyedzafar**
…tions

### CODE REVIEW

File: controller/handler/patient-csv.js [hunk 1]
```diff
@@ -34,6 +38,10 @@ const configuration = [
         label: 'selected answer',
         key: 'answered',
         default: 'false'
+    },
+    {
+        label: 'survey_instance_state',
```
Comment (ChristianMurphy) [inline controller/handler/patient-csv.js]: use spaces instead of underscores in the label

File: controller/handler/patient-csv.js [hunk 2]
```diff
@@ -107,8 +116,12 @@ function patientCSV (request, reply) {
             // send back updated data
             return questionOption;
         });
+        const property = ['id', 'questionText'];
+        const rowsm = deduplicate(allOptionsWithAnswers, property);
+
+        console.log(rowsm);
```
Comment (ChristianMurphy) [inline controller/handler/patient-csv.js]: remove `console.log`

File: controller/handler/patient-csv.js [hunk 3]
```diff
@@ -107,8 +116,12 @@ function patientCSV (request, reply) {
             // send back updated data
             return questionOption;
         });
+        const property = ['id', 'questionText'];
+        const rowsm = deduplicate(allOptionsWithAnswers, property);
```
Comment (ChristianMurphy) [inline controller/handler/patient-csv.js]: change `const allOptionsWithAnswers` to `let allOptionsWithAnswers`
and assign deduplicate to `allOptionsWithAnswers`

File: controller/helper/deduplicate.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ */
+function deduplicate (rows, properties) {
+    const current = {};
+
+    return rows.map((row) => {
+        for (const property of properties) {
+            if (row[property] === current[property]) {
+                row[property] = ' ';
+            } else {
+                current[property] = row[property];
```
Comment (ChristianMurphy) [inline controller/helper/deduplicate.js]: assigning to row directly here will modify the original value passed into `rows`.
instead use `const copyOfRow = Object.create(row);` and modify and return that.

File: controller/handler/patient-csv.js [hunk 5]
```diff
@@ -107,14 +116,16 @@ function patientCSV (request, reply) {
...
+
 
-        return convertJsonToCsv(allOptionsWithAnswers, configuration);
+        return convertJsonToCsv(rowsm, configuration);
     })
     .then((csv) => {
         reply(csv).type('text/csv');
     })
     .catch((err) => {
-        console.error(err);
```
Comment (ChristianMurphy) [inline controller/handler/patient-csv.js]: please keep the `console.error` in the catch statement.

**Comment by shahsyedzafar** [comment]
Yea its there. Sorry I just removed that by mistake.

File: controller/handler/patient-csv.js [hunk 6]
```diff
@@ -107,14 +116,19 @@ function patientCSV (request, reply) {
...
             return questionOption;
         });
+        const property = ['id', 'questionText'];
+        const rowsm = deduplicate(allOptionsWithAnswers, property);
+
+<<<<<<< HEAD
+=======
 
-        return convertJsonToCsv(allOptionsWithAnswers, configuration);
+>>>>>>> 26b4c57280465382507a840b5a7c1b472e7caad1
```
Comment (ChristianMurphy) [inline controller/handler/patient-csv.js]: remove the merge conflict text lines 122-125

File: controller/handler/patient-csv.js [hunk 7]
```diff
@@ -44,43 +52,77 @@ const configuration = [
...
-        ON si.patientId = pa.id
-        JOIN survey_template AS st
-        ON st.id = si.surveyTemplateId
-        JOIN join_surveys_and_questions AS jsq
-        ON jsq.surveyTemplateId = st.id
-        JOIN question_template AS qt
-        ON qt.id = jsq.questionTemplateId
-        JOIN question_option AS qo
-        ON qo.questionTemplateId = qt.id
-        LEFT JOIN question_result AS qr
```
Comment (ChristianMurphy) [inline controller/handler/patient-csv.js]: Keep `LEFT JOIN` instead of two queries and a manual join

**Comment by ChristianMurphy** [comment]
LGTM

**Comment by poojaRal** [comment]
Can you please attach a latest .csv file of how the data gets exported here in a comment. I would like to review the csv file before approving the PR.

**Comment by poojaRal** [comment]
LGTM

**Pull request merged**